plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt

```diff
@@ -492,6 +500,125 @@ private class AmazonQServerInstance(private val project: Project, private val cs
+     * 2. Uses user-specified runtime path from LSP settings if available
+     * 3. Uses system Node.js if version 18+ is available
+     * 4. Falls back to original nodePath with a notification to configure runtime
+     *
+     * @param nodePath The initial Node.js runtime path to check, typically from the artifact directory
+     * @return Path The resolved Node.js runtime path to use for the LSP server
+     *
+     * Side effects:
+     * - Logs warnings if initial runtime path is invalid
+     * - Logs info when using alternative runtime path
+     * - Shows notification to user if no valid Node.js runtime is found
+     *
+     * Note: The function will return a path even if no valid runtime is found, but the LSP server
+     * may fail to start in that case. The caller should handle potential runtime initialization failures.
+     */
+    private fun getNodeRuntimePath(nodePath: Path): Path {
+        if (Files.exists(nodePath) && Files.isExecutable(nodePath)) {
+            return nodePath
+        }
+        // use alternative node runtime if it is not found
+        LOG.warn { "Node Runtime download failed. Fallback to user specified node runtime " }
+        // attempt to use user provided node runtime path
+        val nodeRuntime = LspSettings.getInstance().getNodeRuntimePath()
+        if (!nodeRuntime.isNullOrEmpty()) {
+            LOG.info { "Using node from $nodeRuntime " }
+            return Path.of(nodeRuntime)
+        } else {
+            val localNode = locateNodeCommand()
+            if (localNode != null) {
+                LOG.info { "Using node from ${localNode.toAbsolutePath()}" }
+                return localNode
+            }
+            notifyInfo(
+                "Amazon Q",
+                message("amazonqFeatureDev.placeholder.node_runtime_message"),
+                project = project,
+                listOf(
+                    NotificationAction.create(
+                        message("codewhisperer.actions.open_settings.title")
+                    ) { _, notification ->
+                        ShowSettingsUtil.getInstance().showSettingsDialog(project, message("aws.settings.codewhisperer.configurable.title"))
+                    },
+                    NotificationAction.create(
+                        message("codewhisperer.notification.custom.simple.button.got_it")
+                    ) { _, notification -> notification.expire() }
+                )
+            )
+            return nodePath
+        }
+    }
+
+    /**
+     * Locates node executable using platform-specific command.
+     * Uses 'where' on Windows and 'which' on Unix-like systems.
+     * Only gets node newer than node 18!
+     * @return Path? The absolute path to node if found via where/which, null otherwise
+     */
+    private fun locateNodeCommand(): Path? {
+        val command = if (SystemInfo.isWindows) {
+            arrayOf("where", "node.exe")
+        } else {
+            arrayOf("which", "node")
+        }
```

Comment:
PathEnvironmentVariableUtil.findAllExeFilesInPath?

Reply:
well I cannot import com.intellij.util.PathEnvironmentVariableUtil

Reply:
updated. this seems better than doing which node